### PR TITLE
Dividend History Chart

### PR DESCRIPTION
**Pull request by deeleeramone**
Adds line graph to stocks-->fa-->divs

### CODE REVIEW

File: gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py [hunk 1]
```diff
@@ -460,23 +457,34 @@ def call_divs(self, other_args: List[str]):
...
-            description="Get historical dividends for company",
+            description="Historical dividends for a company",
         )
         parser.add_argument(
             "-n",
             "--num",
             dest="num",
             type=check_positive,
-            default=12,
+            default=20,
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py:467]: 20 is way too much as default IMO. Can we keep this to 10?

File: gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py [hunk 2]
```diff
@@ -460,23 +457,34 @@ def call_divs(self, other_args: List[str]):
...
             help="Number of previous dividends to show",
         )
+        parser.add_argument(
+            "-p",
+            "--plot",
+            dest="plot",
+            default=True,
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py:474]: I think you want default false here

File: gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py [hunk 3]
```diff
@@ -147,15 +143,16 @@ def display_calendar_earnings(ticker: str):
     print("")
 
 
-def display_dividends(ticker: str, num: int = 12, export: str = ""):
+def display_dividends(ticker: str, num: int = 12, plot: bool = False, export: str = ""):
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py:146]: num: int = 10

File: gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py [hunk 4]
```diff
@@ -165,6 +162,31 @@ def display_dividends(ticker: str, num: int = 12, export: str = ""):
...
+        ax.set_title(f"Dividend History for {ticker}")
+        ax.set_xlim(div_history.index[-1], div_history.index[0])
+        if gtff.USE_ION:
+            plt.ion()
+        fig.tight_layout()
+        plt.show()
+
     div_history.index = pd.to_datetime(div_history.index, format="%Y%m%d").strftime(
         "%Y-%m-%d"
     )
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py:192]: let's add an "else" here, in case the user is not plotting

**Review by DidierRLopes — CHANGES_REQUESTED** [review]
@Chavithra do you know why the tests are failing here?

File: gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py (hunk 3 above)
Comment (deeleeramone) [reply gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py:146]: 20 represents five years of dividend history assuming quarterly payments. This is my reasoning for chosing this number.

File: gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py (hunk 1 above)
Comment (deeleeramone) [reply gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py:467]: 20 represents five years of dividend payments, assuming paid quarterly. If this number is still too high, then multiples of four work better here, IMO.

File: gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py (hunk 2 above)
Comment (deeleeramone) [reply gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py:474]: IMO, it's more telling to see the visual, gives a clear signal for strength of a dividend paying company.

**Comment by deeleeramone** [comment]
I’m a newbie at this, so I’m not sure how to implement your ‘else’ request. 🙃

File: gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py (hunk 1 above)
Comment (DidierRLopes) [reply gamestonk_terminal/stocks/fundamental_analysis/fa_controller.py]: That is a fair assumption!

Should we go with 12, 3 years?

I just think that every time I tried having 20 as default it was a bit too much. BUT in this case I was wondering if we should have dividends shown in terms of plot by default? and in that case having 20 is completely fine!

**Comment by DidierRLopes** [comment]
@jmaslek What do you think about having the plot being displayed by default? and instead of having a `plot` options having a `raw` option?

**Comment by jmaslek** [comment]
> @jmaslek What do you think about having the plot being displayed by default? and instead of having a `plot` options having a `raw` option?

Not a fan.  Think the numbers are better to look at then a plot for such small amounts.

**Comment by deeleeramone** [comment]
Updated default to reflect

**Comment by Chavithra** [comment]
Hello @deeleeramone, can you try to run the following commands :

```python
rm -r tests/gamestonk_terminal/stocks/fundamental_analysis/txt/test_fa_controller
pytest tests/gamestonk_terminal/stocks/fundamental_analysis/test_fa_controller.py --record-mode=once
```

It should update the expected result for `test_fa_controller`.

File: gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py [hunk 5]
```diff
@@ -165,6 +162,31 @@ def display_dividends(ticker: str, num: int = 12, export: str = ""):
...
+        )
+        ax.set_xlabel("Date")
+        ax.set_ylabel("Amount ($)")
+        ax.set_title(f"Dividend History for {ticker}")
+        ax.set_xlim(div_history.index[-1], div_history.index[0])
+        if gtff.USE_ION:
+            plt.ion()
+        fig.tight_layout()
+        plt.show()
+
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py:189]: Basically the `else:` should come on this line.

You should have something like:
```
if plot:
   do something for plotting
else:
   just show the raw data
```

Comment (DidierRLopes) [reply gamestonk_terminal/stocks/fundamental_analysis/yahoo_finance_view.py:189]: @deeleeramone can you have a look at this so we can merge it?

**Comment by deeleeramone** [comment]
The file has been updated.

**Comment by DidierRLopes** [comment]
> The file has been updated.
> […](#)

It hasn't yet. You need to "add" then "commit" then "push".

**Comment by DidierRLopes** [comment]
@Chavithra same here happening

**Comment by deeleeramone** [comment]
Do I need to run those two commands again for the test?

**Review by jmaslek — APPROVED** [review]

**Review by DidierRLopes — APPROVED** [review]

**Pull request merged**